brozzler/ydl.py

```diff
@@ -32,9 +32,20 @@
 thread_local = threading.local()
 
 
-def should_ytdlp(page, site):
+def should_ytdlp(worker, site, page):
     # called only after we've passed needs_browsing() check
     if page.status_code != 200:
+        logging.info("skipping ytdlp: non-200 page status")
+        return False
+    if site.skip_ytdlp:
+        logging.info("skipping ytdlp: site marked skip_ytdlp")
+        return False
+
+    ytdlp_seed = site["metadata"]["ait_seed_id"] if "metadata" in site and "ait_seed_id" in site["metadata"] else None
+
+    if ytdlp_seed and ytdlp_seed in worker.skip_av_seeds:
+        logging.info("skipping ytdlp: site in skip_av_seeds")
+        site.skip_ytdlp = True
```

Comment:
Maybe we should also cache the False condition as well, so we avoid the scan of the `skip_av_seeds` list
Then we'd modify the conditional check around line 40 to handle it being True/False or None

Reply:
I prefer simpler code with default = False, I think